File: leetcode/70.爬楼梯.java
/*
 * @lc app=leetcode.cn id=70 lang=java
 *
 * [70] 爬楼梯
 */

// @lc code=start
class Solution {
    public int climbStairs(int n) {
        if (n <= 2) {
            return n;
        }
        int prev = 1, curr = 2, prevprev;

        for (int i = 3; i <= n; i++) {
            prevprev = prev;
            prev = curr;
            curr = prev + prevprev;
        }
        return curr;
    }
}
// @lc code=end

File: leetcode/70.爬楼梯.js
/*
 * @lc app=leetcode.cn id=70 lang=javascript
 *
 * [70] 爬楼梯
 */

// @lc code=start
/**
 * @param {Number} n
 * @return {Number}
 */
var climbStairs = function (n) {
    if (n <= 2) {
        return n
    }
    let memo = {}
    memo[1] = 1
    memo[2] = 2

    for (let i = 3; i <= n; i++) {
        memo[i] = memo[i - 1] + memo[i - 2]
    }
    return memo[n]
}
// @lc code=end
